Shared/Samples/Edit geometries with programmatic reticle tool/EditGeometriesWithProgrammaticReticleToolView.swift

```diff
@@ -0,0 +1,617 @@
+        switch element {
+        case let vertex as GeometryEditorVertex:
+            await mapViewProxy.setViewpoint(
+                Viewpoint(
+                    center: element.extent.center,
+                    scale: Viewpoint.ireland.targetScale
+                ),
+                duration: 0.3
+            )
+            model.geometryEditor.selectVertexAt(
+                partIndex: vertex.partIndex,
+                vertexIndex: vertex.vertexIndex
+            )
+        case let midVertex as GeometryEditorMidVertex:
+            guard model.allowsVertexCreation else { return }
+            
+            await mapViewProxy.setViewpoint(
+                Viewpoint(
+                    center: element.extent.center,
+                    scale: Viewpoint.ireland.targetScale
+                ),
+                duration: 0.3
+            )
+            model.geometryEditor.selectMidVertexAt(
+                partIndex: midVertex.partIndex,
+                segmentIndex: midVertex.segmentIndex
+            )
+        default:
+            break
+        }
+    }
+    
+    /// Starts editing the graphic identified at the tap location.
+    /// - Parameters:
+    ///   - screenPoint: The screen coordinate of the geo view at which to identify.
+    ///   - mapViewProxy: The map view proxy used to identify the screen point.
+    private func startEditingGraphic(screenPoint: CGPoint, mapViewProxy: MapViewProxy) async throws {
+        // Identify graphics in the graphics overlay using the tapped position.
+        let results = try await mapViewProxy.identifyGraphicsOverlays(
+            screenPoint: screenPoint,
+            tolerance: 10
+        )
+        guard let selectedGraphic = results.first?.graphics.first,
+              let geometry = selectedGraphic.geometry else { return }
+        
+        model.startEditing(with: selectedGraphic)
+        
+        // If vertex creation is allowed, set the viewpoint to the center of the selected graphic's geometry.
+        // Otherwise, set the viewpoint to the end point of the first part of the geometry.
+        if model.allowsVertexCreation {
+            await mapViewProxy.setViewpoint(
+                Viewpoint(
+                    center: geometry.extent.center,
+                    scale: Viewpoint.ireland.targetScale
+                ),
+                duration: 0.03
+            )
+        } else {
+            let center: Point?
+            switch model.selectedGraphic?.geometry {
+            case let polygon as Polygon:
+                center = polygon.parts[0].endPoint
+            case let polyline as Polyline:
+                center = polyline.parts[0].endPoint
+            case let multipoint as Multipoint:
+                center = multipoint.points.last
+            case let point as Point:
+                center = point
+            default:
+                return
+            }
```

Comment:
I was a bit hasty with my last suggestion here. This could be even simpler.

```suggestion
            let center: Point? = switch model.selectedGraphic?.geometry {
            case let polygon as Polygon:
                polygon.parts[0].endPoint
            case let polyline as Polyline:
                polyline.parts[0].endPoint
            case let multipoint as Multipoint:
                multipoint.points.last
            case let point as Point:
                point
            default:
                nil
            }
```